Move theming support from Beta to Main

DukeFerdinand opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #121 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [x] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Release version 3.0.0-beta.1 with new custom registered auro-datetime component, theme support, and various performance improvements. Fix rerouted station display issue and update documentation and tests accordingly. Remove GitHub label colors and obsolete scripts.

New Features:
- Introduce custom registered auro-datetime component with tag name as an attribute.

Bug Fixes:
- Fix display issue for rerouted departure and arrival stations.

Enhancements:
- Refactor custom component registration configuration for improved performance.
- Add theme support allowing components to be restyled using design tokens.
- Update dependencies to the latest versions for better performance and compatibility.

Build:
- Remove GitHub label colors from the settings configuration.

Documentation:
- Update documentation to include new attributes and theme support information.

Tests:
- Modify tests to accommodate changes in the auro-datetime component registration.

Chores:
- Remove obsolete scripts related to documentation generation and build preparation.

jordanjones243 committed:
build: update design tokens, eslint config and all other dependencies

jordanjones243 committed:
build: update design tokens, eslint config and all other dependencies

jordanjones243 committed:
feat: implement tier 3 token structure #121

jason-capsule42 committed:
refactor: cleanup previous commit changes that were not necessary

semantic-release-bot committed:
chore(release): 2.7.1 [skip ci]

* **#123:** rerouted departure and arrival stations display as expected now ([5201a33](https://github.com/AlaskaAirlines/auro-flight/commit/5201a337c1abe6e7b0eb05928ac88cbdcaedfe00)), closes [#123](https://github.com/AlaskaAirlines/auro-flight/issues/123)

jordanjones243 committed:
perf: update deps and refactor scripts to use auro-library versions

semantic-release-bot committed:
chore(release): 2.8.0-beta.1 [skip ci]

* implement tier 3 token structure [#121](https://github.com/AlaskaAirlines/auro-flight/issues/121) ([874e9ce](https://github.com/AlaskaAirlines/auro-flight/commit/874e9cedd9d092a1f327bb8d9651fef6eac22a54))

* update deps and refactor scripts to use auro-library versions ([238a929](https://github.com/AlaskaAirlines/auro-flight/commit/238a92976dfa248447e4a11c68b8dbf9ae0b6519))

semantic-release-bot committed:
chore(release): 2.8.0-beta.1 [skip ci]

* implement tier 3 token structure [#121](https://github.com/AlaskaAirlines/auro-flight/issues/121) ([874e9ce](https://github.com/AlaskaAirlines/auro-flight/commit/874e9cedd9d092a1f327bb8d9651fef6eac22a54))

* update deps and refactor scripts to use auro-library versions ([238a929](https://github.com/AlaskaAirlines/auro-flight/commit/238a92976dfa248447e4a11c68b8dbf9ae0b6519))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-flight?pullRequest=127) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-flight?pullRequest=127) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: jason-capsule42<br/>:white_check_mark: jordanjones243<br/>:white_check_mark: DukeFerdinand<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-flight?pullRequest=127) it.</sub>

DukeFerdinand committed:
refactor: post-rebase script additions

Somehow, this repo was missing a lot of the new build scripts, and was not able to generate a new README or other docs.

This commit adds those sans the `build:version` script as we did not need that for this repo.

Changes to be committed:
	modified:   demo/api.html
	modified:   package.json

semantic-release-bot committed:
chore(release): 2.8.0-beta.2 [skip ci]

# [2.8.0-beta.2](https://github.com/AlaskaAirlines/auro-flight/compare/v2.8.0-beta.1...v2.8.0-beta.2) (2024-08-19)

### Bug Fixes

* **#123:** rerouted departure and arrival stations display as expected now ([73eb66c](https://github.com/AlaskaAirlines/auro-flight/commit/73eb66c3d99828b31b0cf9f688c696b14eb0eb1b)), closes [#123](https://github.com/AlaskaAirlines/auro-flight/issues/123)

### Performance Improvements

* update deps and refactor scripts to use auro-library versions ([52683c2](https://github.com/AlaskaAirlines/auro-flight/commit/52683c2da8cbf61611d77b13675a5f877de43ea1))

jordanjones243 committed:
perf: update dependencies

BREAKING CHANGE: trigger major release for color theme support #121

jordanjones243 committed:
perf: remove github label colors

jordanjones243 committed:
feat: add tag name as attribute when custom registered

jordanjones243 committed:
perf: refactor custom component registration config

jordanjones243 committed:
feat: add custom registered auro-datetime

semantic-release-bot committed:
chore(release): 3.0.0-beta.1 [skip ci]

# [3.0.0-beta.1](https://github.com/AlaskaAirlines/auro-flight/compare/v2.8.0-beta.2...v3.0.0-beta.1) (2024-10-03)

### Features

* add custom registered auro-datetime ([33a189b](https://github.com/AlaskaAirlines/auro-flight/commit/33a189b1e866a241d5a5a6fda14effae134270b6))
* add tag name as attribute when custom registered ([26808eb](https://github.com/AlaskaAirlines/auro-flight/commit/26808eb68e4256275954d61c0ba0923bc7517f0d))

### Performance Improvements

* refactor custom component registration config ([1f3e0de](https://github.com/AlaskaAirlines/auro-flight/commit/1f3e0de1444896aef66bc4f1fe75e1ce65e77640))
* remove github label colors ([b5187c0](https://github.com/AlaskaAirlines/auro-flight/commit/b5187c0b4c15de55189e68cdbb7756f234651ad8))
* update dependencies ([2de3a30](https://github.com/AlaskaAirlines/auro-flight/commit/2de3a30d015c11eaedc6178126ccc5b22cf625ab))

### BREAKING CHANGES

* trigger major release for color theme support #121

jordanjones243 committed:
docs: remove beta releases from CHANGELOG

jordanjones243 commented on src/color-flight.scss:
```diff
@@ -0,0 +1,15 @@
+// ---------------------------------------------------------------------
+
+// Import Auro tokens
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables';
```
Not making these changes at this time.

jordanjones243 commented on test/auro-flight-main.test.js:
```diff
@@ -16,8 +16,10 @@ describe('auro-flight-main', () => {
+    console.log("this:", el.shadowRoot.querySelector('.departureTime'));
+
+    await expect(el.shadowRoot.querySelector('.departureTime').querySelector('[auro-datetime]').getAttribute('setDate')).to.equal('2022-05-04T00:30:00-07:00');
+    await expect(el.shadowRoot.querySelector('.arrivalTime').querySelector('[auro-datetime]').getAttribute('setDate')).to.equal('2022-05-04T11:55:00-04:00');
```
Not editing test at this time.

jordanjones243 commented on test/auro-flight-main.test.js:
```diff
@@ -16,8 +16,10 @@ describe('auro-flight-main', () => {
+    await expect(el.shadowRoot.querySelector('.arrivalTime').querySelector('[auro-datetime]').getAttribute('setDate')).to.equal('2022-05-04T11:55:00-04:00');
     await expect(el.shadowRoot.querySelector('.departureStation').textContent.trim()).to.contain('SEA');
     await expect(el.shadowRoot.querySelector('.arrivalStation').textContent.trim()).to.contain('PVD');
   });
```
Not adding new tests at this time.

jordanjones243 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,37 @@
+
+### BREAKING CHANGES
+
+* trigger major release for color theme support #121
```
Not providing more context at this time.

semantic-release-bot committed:
chore(release): 3.0.0-beta.1 [skip ci]

# [3.0.0-beta.1](https://github.com/AlaskaAirlines/auro-flight/compare/v2.7.1...v3.0.0-beta.1) (2024-10-04)

### Features

* add custom registered auro-datetime ([489fb7a](https://github.com/AlaskaAirlines/auro-flight/commit/489fb7ad4ee8023b1d14e7dffae979f8f2833f28))
* add tag name as attribute when custom registered ([fdbbb3b](https://github.com/AlaskaAirlines/auro-flight/commit/fdbbb3b2cf018830eb2a10f96dc137793cc324ce))
* implement tier 3 token structure [#121](https://github.com/AlaskaAirlines/auro-flight/issues/121) ([5149f5d](https://github.com/AlaskaAirlines/auro-flight/commit/5149f5d564cb61cd9ed73ce748fbed918d22aeab))

### Performance Improvements

* refactor custom component registration config ([d0edb86](https://github.com/AlaskaAirlines/auro-flight/commit/d0edb86c5827e5820a15d0ec76866dfb5300efb9))
* remove github label colors ([4b4de54](https://github.com/AlaskaAirlines/auro-flight/commit/4b4de54706ce8cb32cb52d2457b0ba01a2c2366b))
* update dependencies ([a916070](https://github.com/AlaskaAirlines/auro-flight/commit/a916070a742f82415245a41ef390008e189b0db1))
* update deps and refactor scripts to use auro-library versions ([7c53ddd](https://github.com/AlaskaAirlines/auro-flight/commit/7c53dddeb64f3f2a54986d486f5c54e68825e73e))

### BREAKING CHANGES

* trigger major release for color theme support #121
